tests/endpoint-test.js

```diff
@@ -1293,6 +1298,91 @@ describe('Endpoint', function() {
 				});
 			});
 		});
+
+		describe('getRepresentationContent()', function() {
```

Comment:
I'm not entirely sure how the tests work, so I'll stop by and perhaps you can give me a high level overview of what's going on so I can just focus on the important parts.